src/Mvc/Mvc.ApiExplorer/src/ApiResponseTypeProvider.cs

```diff
@@ -228,6 +228,13 @@ private static List<IApiResponseMetadataProvider> GetResponseMetadataAttributes(
 
         foreach (var metadata in responseMetadata)
         {
+            // `IResult` metadata inserted for awaitable types should
```

Comment:
I can't tell from this comment whether `IResult` _only_ appears because we inserted it.  Is there a case where it's present legitimately and we shouldn't skip it?

Reply:
It's theoritically feasible that IResult could appear there if someone manually insert it by doing something like:

```csharp
app.MapGet()
	.Produces<IResult>("application/json");
```

In either case, it still stands that `IResult` doesn't provide us with type information specific enough to include in the response type information.

Reply:
What about something like "`IResult` (usually inserted for awaitable types) isn't useful..."?